libtextsecure/outgoing_message.js

```diff
@@ -293,309 +293,272 @@ OutgoingMessage.prototype = {
+    // }
+    const ttl = getTTLForType(thisDeviceMessageType);
+    const ourKey = textsecure.storage.user.getNumber();
+
+    return {
+      ttl,
+      ourKey,
+      sourceDevice: 1,
+      plaintext,
+      pubKey: devicePubKey,
+      isFriendRequest: enableFallBackEncryption,
+      isSessionRequest,
+      enableFallBackEncryption,
+    };
+  },
 
-        if (window.lokiFeatureFlags.useSealedSender) {
-          const secretSessionCipher = new window.Signal.Metadata.SecretSessionCipher(
-            textsecure.storage.protocol
-          );
-          ciphers[address.getDeviceId()] = secretSessionCipher;
+  async encryptMessage(clearMessage) {
+    if (clearMessage === null) {
+      window.log.warn(
+        'clearMessage is null on encryptMessage... Returning null'
+      );
+      return null;
+    }
+    const {
+      ttl,
+      ourKey,
+      sourceDevice,
+      plaintext,
+      pubKey,
+      isSessionRequest,
+      enableFallBackEncryption,
+    } = clearMessage;
+    // Session doesn't use the deviceId scheme, it's always 1.
+    // Instead, there are multiple device public keys.
+    const deviceId = 1;
+
+    const address = new libsignal.SignalProtocolAddress(pubKey, deviceId);
+
+    let sessionCipher;
+
+    if (enableFallBackEncryption) {
+      sessionCipher = new libloki.crypto.FallBackSessionCipher(address);
+    } else {
+      sessionCipher = new libsignal.SessionCipher(
+        textsecure.storage.protocol,
+        address
+      );
+    }
 
-          const senderCert = new textsecure.protobuf.SenderCertificate();
+    let type;
+    let content;
+    if (window.lokiFeatureFlags.useSealedSender) {
+      const secretSessionCipher = new window.Signal.Metadata.SecretSessionCipher(
+        textsecure.storage.protocol
+      );
+      // ciphers[address.getDeviceId()] = secretSessionCipher;
 
-          senderCert.sender = ourKey;
-          senderCert.senderDevice = deviceId;
+      const senderCert = new textsecure.protobuf.SenderCertificate();
 
-          const ciphertext = await secretSessionCipher.encrypt(
-            address,
-            senderCert,
-            plaintext,
-            sessionCipher
-          );
+      senderCert.sender = ourKey;
+      senderCert.senderDevice = deviceId;
 
-          type = textsecure.protobuf.Envelope.Type.UNIDENTIFIED_SENDER;
-          content = window.Signal.Crypto.arrayBufferToBase64(ciphertext);
-        } else {
-          // TODO: probably remove this branch once
-          // mobile clients implement sealed sender
-          ciphers[address.getDeviceId()] = sessionCipher;
+      const ciphertext = await secretSessionCipher.encrypt(
+        address,
+        senderCert,
+        plaintext,
+        sessionCipher
+      );
+      type = textsecure.protobuf.Envelope.Type.UNIDENTIFIED_SENDER;
+      content = window.Signal.Crypto.arrayBufferToBase64(ciphertext);
+    } else {
+      // TODO: probably remove this branch once
```

Comment:
Now that we always use sealed sender, thoughts on removing all the other branches and simplifying this logic?